### PR TITLE
[SMTChecker] New VariableUsage flag to inline functions

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: libsolidity/formal/BMC.cpp [hunk 1]
```diff
@@ -49,6 +49,8 @@ void BMC::analyze(SourceUnit const& _source, shared_ptr<Scanner> const& _scanner
 
 	m_scanner = _scanner;
 
+	m_variableUsage.setFunctionInlining(true);
```
Comment (chriseth) [inline libsolidity/formal/BMC.cpp:53]: Where is it reset?

**Review by chriseth — DISMISSED** [review]

File: libsolidity/formal/BMC.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/formal/BMC.cpp:53]: Each engine should set it before its execution.
In this case, when added, CHC will set it to false in a similar way.

**Comment by leonardoalt** [comment]
Updated VS compilation problem

File: libsolidity/formal/BMC.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/formal/BMC.cpp:53]: Why not make it part of the constructor then?

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
Conflict.

**Comment by leonardoalt** [comment]
Rebased

**Comment by chriseth** [comment]
Test fails.

**Comment by leonardoalt** [comment]
Needs #7061

**Comment by chriseth** [comment]
Fair enough.

**Pull request merged**